x-pack/plugins/enterprise_search/public/applications/index.tsx

```diff
@@ -67,19 +70,18 @@ export const renderApp = (
           setDocTitle: core.chrome.docTitle.change,
         }}
       >
-        <LicenseProvider license$={plugins.licensing.license$}>
-          <Provider store={store}>
-            <Router history={params.history}>
-              <App {...initialData} />
-            </Router>
-          </Provider>
-        </LicenseProvider>
+        <Provider store={store}>
+          <Router history={params.history}>
+            <App {...initialData} />
+          </Router>
+        </Provider>
       </KibanaContext.Provider>
     </I18nProvider>,
     params.element
   );
   return () => {
     ReactDOM.unmountComponentAtNode(params.element);
+    unmountLicensingLogic();
```

Comment:
If this list gets much longer I might try to think of a better solution than losting all of these inputs.

```
const configs = [ LicenseLogic: {<LicenseLogic props>}, HttpLogic: {<HttpLogic props}, etc ] 
const unmount = Object.entries(config).map((Logic, props) => { Logic(props); return Logic.mount() })

return () => {
  ReactDOM.unmountComponentAtNode(params.element);
  unmounts.forEach(unmount => unmount())
}
```

then we'd never forget to unmount!

Reply:
For sure! FWIW, it stops at 4 total (by the end of my series of PRs) but I can definitely see a case for pulling all our mounts out to a separate file/helper and returning an array of unmounts. I think 4 is just on the border to not a completely warrant a separate helper, but let met know if you disagree 🤔